src/server/pfs/cmds/cmds.go

```diff
@@ -52,7 +53,7 @@ const (
 )
 
 // Cmds returns a slice containing pfs commands.
-func Cmds() []*cobra.Command {
+func Cmds() ([]*cobra.Command, error) {
```

Comment:
why not just panic if we hit an error?

Reply:
Better to return an error and let the caller determine how to handle the error. In this case the [caller](https://github.com/pachyderm/pachyderm/blob/f9a8dc63e8d32a5fbe198ac97de5d02c799d5419/src/server/cmd/pachctl/cmd/cmd.go#L320) _does_ just return an error, but it could return an error too, and print a useful-to-the-user error message.

I could add that to this PR …

Reply:
It seems odd to me that we're only applying it to pfs and pps cmds in this PR - besides updating a potentially preferable signature, I don't think it's providing value in this PR. I would be more in favor of just panicking or refactoring more broadly. panicking seems fine in this context because in a CLI the program terminates anyway.
WDYT?

Reply:
I’d rather refactor more broadly than panic. IMHO a panic indicates a bug, such as a nil memory access. Getting a stack trace can be useful to a developer with a panic; getting a stack trace is useless to en end user. In this case an error would be returned if the user’s `~/.pachyderm/config` has a syntax error — that is not necessarily a bug, but could be caused by the user manually editing the JSON. I think it would be more helpful to print _that_ then a panic trace.

Reply:
Ok I would prefer to leave it out of this Pr, but if you want to commit as is I encourage prioritizing a ticket to finish the job.